<launch>
  <!-- <param name="use_sim_time" value="true"/>   -->
  <node name="map_to_os_fix" pkg="tf2_ros" type="static_transform_publisher" args="0 0 0 0 0 0 map os_sensor"/>

  <node name="scans_demo" pkg="rosbag" type="play"  args="--clock --loop $(find lidar_obstacle_detection)/resources/scans_demo.bag"/>

  <node name="nodelet_manager" pkg="nodelet" type="nodelet" args="manager" output="screen">
    <param name="num_worker_threads" value="20"/>
  </node>


  <node name="detect" pkg="lidar_obstacle_detection" type="main" output="screen">
    
  </node>

  <!-- <node name="detect" pkg="nodelet" type="nodelet" args="load lidar_obstacle_detection/Detect nodelet_manager">
    <param name="active"            value="true"/>
    <param name="publish_scan"      value="false"/>
    <param name="publish_pcl"       value="true"/>

    <param name="ranges_num"        value="1000"/>

    <param name="min_scanner_range" value="0.05"/>
    <param name="max_scanner_range" value="10.0"/>

    <param name="min_x_range"       value="-10.0"/>
    <param name="max_x_range"       value="10.0"/>
    <param name="min_y_range"       value="-10.0"/>
    <param name="max_y_range"       value="10.0"/>

    <param name="fixed_frame_id"    value="map"/>
    <param name="target_frame_id"   value="robot"/>
  </node> -->

  <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find lidar_obstacle_detection)/resources/lidar_obstacle_detector.rviz"/>

</launch>
